advanced-usage.md

```diff
@@ -325,6 +325,20 @@ for (std::future<cpr::Response>& fr: container) {
 ```
 {% endraw %}
 
+Asynchronous requests can also be performed using a `cpr::Session` object. It is important to note that the asynchronous request is performed directly on the session object, modifying it in the process.
+To ensure that the lifetime of the session is properly extended, the session object used must be managed by a `std::shared_ptr`. Here is an example for an asynchronous get request which uses a session object:
+
+{% raw %}
+```c++
+std::shared_ptr<cpr::Session> session = std::make_shared<cpr::Session>();
+cpr::Url url = cpr::Url{"http://www.httpbin.org/get"};
+session->SetUrl(url);
+cpr::AsyncResponse fr = session->GetAsync();
+cpr::Response r = fr.get();
```

Comment:
```suggestion
cpr::Response r = asyncRresp.get();
```

`fr` is not descriptive in my eyes. Why "fr"?

Reply:
I have just copied the name "fr" from the other examples of the same docs chapter as I wanted to stay consistent. I think "fr" stands for future.
However, I also think it's not very descriptive and will fix this.

Reply:
Aha! Then we can keep it and I will update it later in a separate PR.